crates/router/src/core/payment_methods/network_tokenization.rs

```diff
@@ -0,0 +1,700 @@
+        .await
+        .change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed);
+    let res = response
+        .map_err(|error| {
+            error.change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed)
+        })
+        .attach_printable("Error while receiving response")
+        .and_then(|inner| match inner {
+            Err(err_res) => {
+                let parsed_error: NetworkTokenErrorResponse = err_res
+                    .response
+                    .parse_struct("Delete Network Tokenization Response")
+                    .change_context(
+                        errors::NetworkTokenizationError::ResponseDeserializationFailed,
+                    )?;
+                logger::error!(
+                    error_code = %parsed_error.error_info.code,
+                    developer_message = %parsed_error.error_info.developer_message,
+                    "Network tokenization error: {}",
+                    parsed_error.error_message
+                );
+                Err(errors::NetworkTokenizationError::ResponseDeserializationFailed)
+                    .attach_printable(format!("Response Deserialization Failed: {err_res:?}"))
+            }
+            Ok(res) => Ok(res),
+        })
+        .inspect_err(|err| {
+            logger::error!("Error while deserializing response: {:?}", err);
+        })?;
+
+    let check_token_status_response: CheckTokenStatusResponse = res
+        .response
+        .parse_struct("Delete Network Tokenization Response")
+        .change_context(errors::NetworkTokenizationError::ResponseDeserializationFailed)?;
+
+    match check_token_status_response.payload.token_status {
+        TokenStatus::Active => Ok((
+            Some(check_token_status_response.payload.token_expiry_month),
+            Some(check_token_status_response.payload.token_expiry_year),
+        )),
+        TokenStatus::Inactive => Ok((None, None)),
+    }
+}
+
+pub async fn delete_network_token_from_locker_and_token_service(
+    state: &routes::SessionState,
+    customer_id: &id_type::CustomerId,
+    merchant_id: &id_type::MerchantId,
+    payment_method_id: String,
+    network_token_locker_id: Option<String>,
+    network_token_requestor_reference_id: String,
+) -> errors::RouterResult<DeleteCardResp> {
+    //deleting network token from locker
+    let resp = payment_methods::cards::delete_card_from_locker(
+        state,
+        customer_id,
+        merchant_id,
+        network_token_locker_id
+            .as_ref()
+            .unwrap_or(&payment_method_id),
+    )
+    .await?;
+    if delete_network_token_from_tokenization_service(
+        state,
+        network_token_requestor_reference_id,
+        customer_id,
+    )
+    .await
+    .is_ok()
+    {
+        logger::info!("Token From Tokenization Service deleted Successfully!");
+    } else {
+        logger::error!("Error while deleting Token From Tokenization Service!");
+    }
```

Comment:
Won't it be better to log the error in case of failures? We can consider logging the error, as long as it does not contain any sensitive information.

Reply:
we are logging the error inside the function - `delete_network_token_from_tokenization_service`